common/hyperscaler/rules/signature_set.go

```diff
@@ -0,0 +1,30 @@
+package rules
+
+type SignatureSet struct {
+	items map[string][]*ParsingResult
+}
+
+func NewSignatureSet(results []*ParsingResult) *SignatureSet {
+	signatureSet := &SignatureSet{
+		items: make(map[string][]*ParsingResult),
+	}
+
+	for _, result := range results {
+		if result.HasErrors() {
+			continue
+		}
+
+		signature := result.Rule.SignaturePreKeys(ASTERISK, ATTRIBUTE_WITH_VALUE)
+		signatureSet.items[signature] = append(signatureSet.items[signature], result)
+	}
+
+	return signatureSet
+}
+
+func (s *SignatureSet) ContainsMirrored(rule *Rule) []*ParsingResult {
```

Comment:
The name `contains` is misleading. By conventions is used for predicates mostly.
What do we return if there are no mirroredSignatureItems?
Empty slice?

Reply:
https://github.com/ralikio/kyma-environment-broker/commit/42071b113a62a4c71ffb3fd7ff4bee932a43d4a7